pilot/pkg/networking/networking.go

```diff
@@ -204,3 +221,64 @@ func MessageToAny(msg proto.Message) *anypb.Any {
 	}
 	return out
 }
+
+func BuildCatchAllVirtualHost(allowAnyoutbound bool, sidecarDestination string) *route.VirtualHost {
+	if allowAnyoutbound {
+		egressCluster := PassthroughCluster
+		notimeout := durationpb.New(0)
+
+		if sidecarDestination != "" {
+			// user has provided an explicit destination for all the unknown traffic.
+			// build a cluster out of this destination
+			egressCluster = sidecarDestination
+		}
+
+		routeAction := &route.RouteAction{
+			ClusterSpecifier: &route.RouteAction_Cluster{Cluster: egressCluster},
+			// Disable timeout instead of assuming some defaults.
+			Timeout: notimeout,
+			// Use deprecated value for now as the replacement MaxStreamDuration has some regressions.
+			// nolint: staticcheck
+			MaxGrpcTimeout: notimeout,
+		}
+
+		// Register catchall virtual host with merge logic, so that it clones while merging.
+		return merge.MustClone(&route.VirtualHost{
```

Comment:
If we clone this each time. how could it be efficient?

Reply:
It does not clone every time - it is only cloned if Envoy filter has virtualhost with allow_any that means a user is making change. So he will pay the price. Others it will be efficient.

Reply:
I am still confused about the judgment. And not understand the change to merge at all. 

So I am concerned about two points:

1. the proto merge is already too complex, should we encourage making it more complex and even harder to maintain. I do not think so.
2. buildCatchAllVirtualHost is just constructing a simple struct with almost no inferring, I do not think it is a performance hot point.

Reply:
> buildCatchAllVirtualHost is just constructing a simple struct with almost no inferring, I do not think it is a performance hot point.

See my point below. This is not just built for catchAllVirtualHost. I agree it is not a huge deal to build it every time. But this is meant to build a generic solution and prove that it works with this.

Reply:
see point 1

Reply:
> the proto merge is already too complex, should we encourage making it more complex and even harder to maintain. I do not think so.

IMO, at some point we have to take this hit because of envoy filters. We can not loose obvious optimizations (for some it may be very time consuming to build), we may need this complexity. Not sure if there is any simpler way to get both the things - optimzation + envoy filter working

Reply:
I suggest we make it when we met some real performance issue.  This does not bring much benefit other than complexity to me.

Reply:
fair point